app/src/main/java/tech/ula/MainActivity.kt

```diff
@@ -565,8 +565,12 @@ class MainActivity : AppCompatActivity(), SessionListFragment.SessionSelection,
             customDialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener {
                 customDialog.dismiss()
                 val sshTypePreference = customDialog.find<RadioButton>(R.id.ssh_radio_button)
-                val selectedPreference =
-                        if (sshTypePreference.isChecked) SshTypePreference else VncTypePreference
+                val vncTypePreference = customDialog.find<RadioButton>(R.id.vnc_radio_button)
```

Comment:
I think it's worth being explicit here, instead of just using an `else` branch to default to Xsdl. I don't know if the `else` branch could ever be hit there, but just in case, error checking would then need to be added to `AppsStartupFsm#setAppServicePreference`.

Reply:
I agree, I've added another variable for xsdlTypePreference here to be more explicit